source/common/amqpClient.js

```diff
@@ -25,646 +37,980 @@ const MONITOR_EXC = {
+  constructor(amqpCli, onMessage) {
+    super(amqpCli, MONITOR_EXC, getQueueName("monitor", amqpCli.idx));
+    this.onMessage = onMessage;
+    this.log = logger.getLogger('Monitor');
+    this.subscriptions.set('monitor', {
+      patterns: ['exit.#'],
+      cb: this.receiveMsg.bind(this)
+    });
   }
 
-  notify(reason, message) {
-    const channel = this.bus.channel;
-    if (this.ready && channel) {
-      const pattern = 'exit.' + reason;
-      const content = JSON.stringify({reason, message});
-      try {
-        channel.publish(
-            MONITOR_EXC.name, pattern, Buffer.from(content));
-      } catch (e) {
-        log.warn('Failed to publish:', e);
-      }
-    } else {
-      this.ready = false;
+  receiveMsg(msg, routingKey) {
+    let self = this;
+    try {
+      msg = JSON.parse(msg.toString());
+      self.log.info('Received monitoring message:', msg);
+      self.onMessage && self.onMessage(msg);
+    } catch (error) {
+      this.log.error('Error processing monitored message:', msg, 'and error:', util.inspect(error));
     }
   }
+}
 
-  close() {
-    this.ready = false;
-    // return this.channel.deleteExchange(MONITOR_EXC.name, {ifUnused: true})
-    //   .catch((err) => log.error('Failed to delete exchange:', this.name));
+class MonitoringTarget extends MsgSender {
+  constructor(amqpCli) {
+    super(amqpCli, MONITOR_EXC);
+    this.log = logger.getLogger('MonitoringTarget');
+  }
+
+  notify(reason, message) {
+    try {
+      this.publish(`exit.${reason}`, {reason, message});
+    } catch (e) {
+      this.log.warn('Failed to publish:', e);
+    }
   }
 }
 
-const RECONNECT = true;
-const RECONNECT_INTERVAL = 1000;
+class ConnDesc {
+  constructor(host, amqpCli) {
+    this.hostname = host;
+    this.isNormal = true;                   //when connect succ it will be true
+    this.isConnected = false;
+    this.isStart = false;                   //when start connect it will be true
+    this.rcCount = 0;
+    this.rcLock = false;                    //reconnect locker
+    this.conn = amqpCli.connection;
+    this.ch = amqpCli.channel;
+    this.amqpCli = amqpCli;
+  }
+
+  onStart() {
+    this.isStart = true;
+    log.info(`${this.amqpCli.idx} connect to ${this.hostname} start`);
+  };
+
+  async onConnected(conn) {
+    log.info(`${this.amqpCli.idx} connect to ${this.hostname} ok`);
+    this.isConnected = true;
+    this.isNormal = true;
+    this.conn = conn;
+  };
+
+  async onChannelOk(ch) {
+    log.info(`${this.amqpCli.idx} onChannelOk ${this.hostname} ok`);
+    this.ch = ch;
+  };
+
+  async onSetupOk() {
+    log.info(`${this.amqpCli.idx} onSetupOk ${this.hostname} ok`);
+    this.rcLock = false;
+  };
+
+  tryReconnect() {
+    if (this.rcLock) {
+      return false;
+    }
+    this.rcLock = true;
+    this.rcCount++;
+    return true;
+  };
+
+  wait(t) {
+    return new Promise((resolve) => setTimeout(resolve, t));
+  };
+
+  async waitToReconnect() {
+    await this.wait(5000);
```

Comment:
Reconnect timeout 5000?

Reply:
when we discover anomalies in the rabbitmq link, we reconnect it every 5 seconds to avoid excessive TCP links penetrating into rabbitmq and causing greater pressure on it